daemon/container_operations_unix.go

```diff
@@ -999,6 +999,9 @@ func (daemon *Daemon) releaseNetwork(container *container.Container) {
 
 	sid := container.NetworkSettings.SandboxID
 	settings := container.NetworkSettings.Networks
+	if container.NetworkSettings.Ports != nil {
```

Comment:
Maybe just set `container.NetworkSettings.Ports = nil` no matter what and avoid the if branch.